### PR TITLE
Add `@metamask/auto-changelog`

### PR DESCRIPTION
**Pull request by Gudahtt**
A basic empty CHANGELOG.md has been added to the project, along with `@metamask/auto-changelog` to help validate and keep the changelog up-to-date.

A step has been added to the "Build, Lint, and Test" workflow to validate the changelog.

### CODE REVIEW

File: .github/workflows/build-test.yml
Comment (Gudahtt) [inline .github/workflows/build-test.yml:38]: Should probably change this :sweat_smile: 

Or maybe that can wait for the PR that adds the release automation

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]: I think the release automation actions perform this validation, so maybe this step is unnecessary?

Comment (Gudahtt) [reply .github/workflows/build-test.yml:38]: Huh. I thought we had actions for creating a release PR and releasing upon the merge of that PR. This step is to validate the release PR before it's merged. I didn't know that was covered in either of the release actions.

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]: Well, `create-release-pr` updates the changelog, and it seems to error if the changelog is invalid as well as `validateChangelog`: https://github.com/MetaMask/action-create-release-pr/blob/c220550e1faca0618d22b755dc8b1b4474473f17/src/package-operations.ts#L186

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]: Hm, actually maybe not. That's certainly easy to add in that Action though. Not sure if we prefer it one place over the other? Here is fine, we should just update the prefix to `release/`.

Comment (Gudahtt) [reply .github/workflows/build-test.yml:38]: But that doesn't get run on subsequent commits on the release PR though. This is run on all release PR commits, and the `--rc` flag additionally ensures the manual categorization is done. It doesn't sound like it does the same thing at all :thinking:

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]: I was referring to RC validation specifically. If we did that in `create-release-pr`, it would obviate the need for knowing the release branch prefix here. But, I'm agnostic!

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]:
```suggestion
        if: ${{ startsWith(github.ref, 'release/') }}
```

File: .github/workflows/build-test.yml
Comment (rekmarks) [inline .github/workflows/build-test.yml:41]:
```suggestion
        if: ${{ !startsWith(github.ref, 'release/') }}
```

File: .github/workflows/build-test.yml
Comment (Gudahtt) [reply .github/workflows/build-test.yml:38]: Right, it still doesn't sound like we're on the same page about what this actually does though. `create-release-pr` _can't_ do that, because `--rc` here also ensures that changes have been categorized. That has to be done manually _after_ the release PR has been created.

Comment (rekmarks) [reply .github/workflows/build-test.yml:38]: Ah, I didn't realize it checks categorization. Thanks for clarifying!

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**
